Backend/router/v1/report.ctrl.js

```diff
@@ -1,51 +1,69 @@
 const models = require('../../models');
 
 const createReport = (req, res) => {
-  models.report.create({
-    what: req.body.data[0].what,
-    location: req.body.data[0].location,
-    time: req.body.data[0].time,
-    who: req.body.data[0].who,
-    details: req.body.data[0].details,
-  }).then((result) => {
-    res.status(201);
-    res.send("good");
-  }).catch((err) => {
-    res.json({msg: "error"});
-  });
-}
+  if(req.isAuthenticated()){
+    models.report.create({
+      what: req.body.data[0].what,
+      location: req.body.data[0].location,
+      time: req.body.data[0].time,
+      who: req.body.data[0].who,
+      details: req.body.data[0].details,
+      userID: req.user[0].dataValues.id
+    }).then((result) => {
+      res.status(201);
+      res.json({
+        msg: "Report is made successfully"
+      });
+    });
+  } else {
+    res.status(403);
+    res.json({
+      msg : "You are not logged in"
+    });
+  }
+};
 
 const showReportList = (req, res) => {
-  models.report.findAll().then((result) => {
-    res.json(result);
-  }).catch((err) => {
-    res.json({msg: "error"});
-  });
-}
+  if(req.isAuthenticated()){
+    models.report.findAll({
+      where: {
+        userID: req.user[0].dataValues.id
+      }
+    }).then((result) => {
+      res.status(200);
+      res.json(result);
+    }).catch((err) => {});
+  } else{
+    res.status(401)
+    res.json({
+      msg: "get report failed"
```

Comment:
This is retrieving all reports, right? If so, 'Failed to get all reports' or something along this line might be better.

Reply:
I'm so sorry for the trouble, Soo. I modified it in my branch.